python/vyos/ifconfig/interface.py

```diff
@@ -336,8 +366,26 @@ def __init__(self, ifname, **kargs):
         self.operational = self.OperationalClass(ifname)
         self.vrrp = VRRP(ifname)
 
+    def get_interface_within_netns(self, key):
```

Comment:
Why are this helpers required? Why can't the netns functionality be added into the base class handler of

https://github.com/vyos/vyos-1x/blob/current/python/vyos/ifconfig/control.py#L172-L177 ?

When netns is assigned in `__init__` one can always check netns in e.g. `vyos.ifconfig.control.get_interface()` and perform the netns relevant functions. No upper layer code needs to be adjusted as everything is handled inside this method.

So if netns is specified - all yout commands will get the `ip netns exec` prefix which can be generalised inside the base classes.

Reply:
Not all commands should be executed in netns
At least this command should be executed from default netns https://github.com/vyos/vyos-1x/blob/aef116371f786f5d711731fdc712ae83dc4a34e2/python/vyos/ifconfig/interface.py#L140-L142

So `exec_within_netns` commands should somehow be executed from control `_get_command` and `_set_command` ?
https://github.com/vyos/vyos-1x/blob/aef116371f786f5d711731fdc712ae83dc4a34e2/python/vyos/ifconfig/control.py#L54-L59

for example `def _get_command(self, config, name, netns=''):` ?

Reply:
No need to change the signature. self._config[“netns“] has the name as the co fog is copied in __init__

Reply:
> At least this command should be executed from default netns

We need to execute this from the *old* netns if moving an interface from a->b where a is not the default netns? (Talking theoretically here, I haven't tried this)